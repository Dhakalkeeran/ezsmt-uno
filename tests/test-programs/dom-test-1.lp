#theory lp {
    linear_term {
    - : 2, unary;
    * : 1, binary, left;
    + : 0, binary, left;
    - : 0, binary, left
    };

    dom_term {
    - : 3, unary;
    + : 3, unary;
    * : 2, binary, left;
    + : 1, binary, left;
    - : 1, binary, left;
    .. : 0, binary, left
    };

    &dom/0 : dom_term, {=}, linear_term, head;
    &logic/1 : linear_term, head
}.

&logic(lia).
node(0).
&dom {(-8)..(-6); (-2)..0; 5; 9..11} = at(X) :- node(X).
