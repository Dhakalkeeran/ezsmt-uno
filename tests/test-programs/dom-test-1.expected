node(0) 
at(0)=(- 8) 

node(0) 
at(0)=(- 7) 

node(0) 
at(0)=(- 6) 

node(0) 
at(0)=(- 2) 

node(0) 
at(0)=(- 1) 

node(0) 
at(0)=0 

node(0) 
at(0)=9 

node(0) 
at(0)=10 

node(0) 
at(0)=11 

node(0) 
at(0)=5
